src/data_source/storage/sql.rs

```diff
@@ -1781,4 +1953,105 @@ mod test {
         assert_eq!(cfg.host, "localhost");
         assert_eq!(cfg.port, 5432);
     }
+
+    #[async_std::test]
+    async fn test_target_period_pruning() {
+        setup_test();
+
+        let db = TmpDb::init().await;
+        let port = db.port();
+        let host = &db.host();
+
+        let migrations = include_migrations!("$CARGO_MANIFEST_DIR/migrations").collect::<Vec<_>>();
+        let cfg = Config::default()
+            .user("postgres")
+            .password("password")
+            .host(host)
+            .port(port)
+            .migrations(migrations);
+
+        let mut storage = SqlStorage::connect(cfg).await.unwrap();
+        let mut leaf = LeafQueryData::<MockTypes>::genesis(&TestInstanceState {});
+        // insert some mock data
+        for i in 0..20 {
+            leaf.leaf.block_header.block_number = i;
+            storage.insert_leaf(leaf.clone()).await.unwrap();
+            storage.commit().await.unwrap();
+        }
+
+        // Set pruner config to default which has minimum retention set to 1 day
+        storage.set_pruning_config(PrunerCfg::new());
+        sleep(Duration::from_secs(2)).await;
+        // No data will be pruned
+        storage.prune().await.unwrap();
+        // Pruned height should be none
+        assert!(storage.pruned_height().is_none());
+
+        // Set pruner config to target retention set to 1s
+        storage.set_pruning_config(PrunerCfg::new().with_target_retention(Duration::from_secs(1)));
+        // All of the data is now atleast one sec old
+        storage.prune().await.unwrap();
+
+        // Pruned height should be some
+        // the table should now contain only 1 row as data with height < target height is pruned
```

Comment:
What is the 1 row? I thought all data was older than 1s by this point?

Reply:
changed the check when deleting rows to make it inclusive. Count is number of rows and  == 0